src/main/java/com/contentgrid/gateway/runtime/RuntimeConfiguration.java

```diff
@@ -45,12 +49,25 @@
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnProperty(value = "contentgrid.gateway.runtime-platform.enabled")
 public class RuntimeConfiguration {
+    @Bean
+    RuntimeDeploymentGatewayFilter deploymentGatewayFilter() {
+        return new RuntimeDeploymentGatewayFilter();
+    }
 
+    @Bean
+    RouteLocator runtimeAppRouteLocator(RouteLocatorBuilder builder, RuntimeRequestResolver requestResolver) {
+        return builder.routes()
+                .route(r -> r
```

Comment:
Couldn't this be a plain gateway configuration file? Something like

That would make it easier to add additional routes that do not forward to a specific application, and makes it easier to configure additional filters when that is necessary

```yaml
spring:
  cloud:
    gateway:
      routes:
        - id: apps
          uri: cg://ignored
          predicates:
            - ResolvesToApplication
          filters:
            - PreserveHostHeader
```

In that case, we would need to explicitly define a `ResolvesToApplication` gateway predicate, but that's all.

Reply:
This could be a static configuration via properties, yes.

1. Given that `spring.cloud.gateway.routes` is a list, it is imo too easy to overwrite a route-definition by accident 
2. makes runtime integration tests more finicky - now the behaviour is enabled with a single flag `contentgrid.gateway.runtime-platform.enabled: true`

This can be revisited in the future.